pandas/tests/test_strings.py

```diff
@@ -435,6 +435,12 @@ def test_replace(self):
                 for data in (['a', 'b', None], ['a', 'b', 'c', 'ad']):
                     values = klass(data)
                     self.assertRaises(TypeError, values.str.replace, 'a', repl)
+        
+        # GH 15055, callable repl
+        repl = lambda m: m.group(0).swapcase()
+        result = values.str.replace('[a-z][A-Z]{2}', repl, n=2)
+        exp = Series([u('foObaD__baRbaD'), NA])
```

Comment:
Thanks, @jreback . Named field/group is a good idea.

If the callable raises a `TypeError` because of a wrong number of arguments, this is caught in [`pandas.core.strings._map`](https://github.com/hzpc-joostk/pandas/blob/pandas-GH15055-patch-1/pandas/core/strings.py#L175). 

We could overcome this by either inspecting the callable signature: `inspect.signature` or trying with a dummy Regex Match object. But this might have the side effect that the callable is called once before doing the replace on the array, confusing users with some kind of counter within the callable.

Sorry about the checks going wrong. It seems I am too careless with checking my commits. 😕